storage: increase default for MaxConcurrentDownloads

stevendanna opened the pull request:
Since cockroachdb/pebble#3445 the concurrency configuration for download compactions is now independent from normal compactions. As a result, our default configuration was only allowing a single download compaction at a time. This has a rather dramatic impact on overall download phase throughput for online restore.

This reclaims some of our throughput but we still have a problem somewhere as our roachtests now take 2-3 times as long (although with this they at least pass before the 1 hour timeout).

Fixes #121265
Fixes #121266

Release note: None

stevendanna committed:
storage: increase default for MaxConcurrentDownloads

Since cockroachdb/pebble#3445 the concurrency configuration for
download compactions is now independent from normal compactions. As a
result, our default configuration was only allowing a single download
compaction at a time. This has a rather dramatic impact on overall
download phase throughput for online restore.

This reclaims some of our throughput we still have a problem somewhere
as our roachtests now take 2-3 times as long (although with this they
at least pass before the 1 hour timeout).

Fixes #121265
Fixes #121266

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/121271)
<!-- Reviewable:end -->

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/121271#-:-Nu43izDDG26NMH7GZSJ:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @jbowens)



<!-- Sent from Reviewable.io -->

stevendanna committed:
storage: make MaxConcurrentDownloads a cluster setting

Since this is IO bound, we don't care too much about GOMAXPROCs like
we do for MaxConcurrentCompactions. Instead, we convert this to a
simple cluster setting.

Release note: None
Epic: none

stevendanna commented:
bors r=dt,RaduBerinde
